### PR TITLE
[FLINK-28008] Can not get secondary resource from context after operator restart

### PR DESCRIPTION
**Pull request by morhidi**
- bumping JOSDK version v3.0.2 -> v3.0.3
- add primaryToSecondary mapper for session jobs
- removing the workaround for setting namespaces (not needed anymore from JOSDK v3.0.3+)
- Changing config dumps into a one-liner

### CODE REVIEW

**Comment by morhidi** [comment]
cc @Aitozi

File: flink-kubernetes-operator/src/main/resources/log4j2.properties [hunk 1]
```diff
@@ -24,3 +24,6 @@ appender.console.name = ConsoleAppender
 appender.console.type = CONSOLE
 appender.console.layout.type = PatternLayout
 appender.console.layout.pattern = %style{%d}{yellow} %style{%-30c{1.}}{cyan} %highlight{[%-5level] [%X{resource.namespace}.%X{resource.name}] %msg%n%throwable}
+
+logger.conf.name = org.apache.flink.configuration.GlobalConfiguration
+logger.conf.level = WARN
```
Comment (gyfora) [inline flink-kubernetes-operator/src/main/resources/log4j2.properties:29]: Why did you change this? Isn't it helpful to see what exact config is loaded by the operator?

Comment (gyfora) [reply flink-kubernetes-operator/src/main/resources/log4j2.properties:29]: makes sense :)

**Review by gyfora — APPROVED** [review]
Looks good, will merge this after the config log fix
